### PR TITLE
Rename android arch lifecycle artifacts to androidx lifecycle naming

### LINKED ISSUE #379: 2.x: Rename android arch components artifacts
These can be renamed to just `autodispose-android-lifecycle` since they're really just androidx lifecycle extensions

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #379 

Will merge manually to best-effort protect git history

### CODE REVIEW

File: android/autodispose-androidx-lifecycle-test/gradle.properties [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-POM_NAME=AutoDispose (Android Architecture Components Extensions)
-POM_ARTIFACT_ID=autodispose-android-archcomponents
+POM_NAME=AutoDispose (androidx-lifecycle test extensions)
```
Comment (ShaishavGandhi) [inline android/autodispose-androidx-lifecycle-test/gradle.properties]: My capitalizing OCD has trouble with `test extensions`

Comment (ZacSweers) [reply android/autodispose-androidx-lifecycle-test/gradle.properties]: Sorry, went for consistency here

Comment (ZacSweers) [reply android/autodispose-androidx-lifecycle-test/gradle.properties]: "Androidx-Lifecycle" would be more wrong

Comment (ShaishavGandhi) [reply android/autodispose-androidx-lifecycle-test/gradle.properties]: Yeah i know. Not suggesting we change, but just had to point out :D

**Pull request merged**